soc/xtensa/intel_adsp/tools/cavstool.py

```diff
@@ -329,6 +484,45 @@ def ipc_command(data, ext_data):
         ext_data = t - ipc_timestamp
         ipc_timestamp = t
         send_msg = True
+    elif data == 5: # HDA RESET (init if not exists)
+        stream_id = ext_data & 0xff
+        if stream_id in hda_streams:
+            hda_streams[stream_id].reset()
+        else:
+            hda_str = HDAStream(stream_id)
+            hda_streams[stream_id] = hda_str
+    elif data == 6: # HDA CONFIG
+        stream_id = ext_data & 0xFF
+        buf_len = ext_data >> 8 & 0xFFFF
+        hda_str = hda_streams[stream_id]
+        hda_str.config(buf_len)
+    elif data == 7: # HDA START
+        stream_id = ext_data & 0xFF
+        hda_streams[stream_id].start()
+
+    elif data == 8: # HDA STOP
+        stream_id = ext_data & 0xFF
+        hda_streams[stream_id].stop()
+    elif data == 9: # HDA VALIDATE
+        stream_id = ext_data & 0xFF
+        hda_str = hda_streams[stream_id]
+        hda_str.debug()
+        is_ramp_data = True
+        hda_str.mem.seek(0)
+        for (i, val) in enumerate(hda_str.mem.read(256)):
+            if i != val:
+                is_ramp_data = False
+            log.info("stream[%d][%d]: %d", stream_id, i, val)
+        log.info("Is ramp data? " + str(is_ramp_data))
+        ext_data = int(is_ramp_data)
+        log.info(f"Ext data to send back on ramp status {ext_data}")
+        send_msg = True
+    elif data == 10: # HDA HOST OUT SEND
```

Comment:
Similarly, you could have a "write this word to this stream/offset" command and avoid the problem of having to synchronize the buffer formats between C and python.